internal/completions/types/types.go

```
@@ -107,11 +107,62 @@ func (p *CompletionRequestParameters) Attrs(feature CompletionsFeature) []attrib
 }
 
 type CompletionResponse struct {
-	Completion string    `json:"completion"`
-	StopReason string    `json:"stopReason"`
+	// Completion is the full completion string.
+	// In the V2 API, this field is empty for streaming responses.
+	// Use DeltaText instead.
+	Completion string `json:"completion,omitempty"`
+	// DeltaText is the incremental text that was added to the prompt.
```

I'd strongly suggest that we don't "overload" the meaning of these fields, as it makes it much more difficult to reason about.

Adding fields with the text like "is only set when API version is higher than X" is one thing. But if you are adding AND removing fields -- i.e. making a breaking API change -- then we should just introduce a separate data type.

e.g. `type CompletionResponseV1 struct` and `typeCompletionResponseV2 struct`. And having the completion provider return one over the other. Or -- have completion providers return some "generic" response type, and then have an adapter that will convert it into the V1 or V2 type as appropriate.

This just seems like something that might lead to confusion and bugs, and so anything we can do to be more explicit and/or "either work correctly or fail loudly" would be preferable, IMHO.